### PR TITLE
T38051 Update "pipeline-details" for timeout and tarball

### PR DESCRIPTION
**Pull request by JenySadadia**
- Update pipeline flow diagram for timeout service.
- Since we have merged checkout and tarball node into a single checkout node, need to update the documentation accordingly.

### CODE REVIEW

**Comment by JenySadadia** [comment]
Documentation on staging:
https://static.staging.kernelci.org/docs/api/pipeline-details/

File: doc/pipeline-details.md [hunk 1]
```diff
@@ -15,15 +15,14 @@ flowchart
     subgraph trigger_service[Trigger Service]
         trigger[Trigger] --> kernel_revision{New kernel <br/>revision ?}
         kernel_revision --> |No| trigger
```
Comment (gctucker) [inline doc/pipeline-details.md:17]:
```
|No| trigger
```
What does `trigger` mean here?  I think it should be something more like `sleep`.

File: doc/pipeline-details.md [hunk 2]
```diff
@@ -15,15 +15,14 @@ flowchart
...
-        kernel_revision --> |Yes| checkout[Create 'checkout' node <br />state=available, result=None, set holdoff]
+        kernel_revision --> |Yes| checkout[Create 'checkout' node <br />state=running, result=None]
     end
     subgraph tarball_service[Tarball Service]
-        checkout --> |event: <br />checkout created, state=available| tarball[Tarball]
-        tarball --> tarball_node[Create 'tarball' node <br />state=running, result=None, holdoff=None <br />Update 'checkout' node <br />with describe and artifacts]
-        tarball_node --> upload_tarball[Create and upload tarball to the storage <br />state=available, result=None, set holdoff]
+        checkout --> |event: <br />checkout created, state=running| tarball[Tarball]
```
Comment (gctucker) [inline doc/pipeline-details.md:21]: What does `tarball[Tarball]` mean here?  I think the block from the `runner` service about uploading the tarball should be moved here instead as that's what the tarball service does.

Comment (JenySadadia) [reply doc/pipeline-details.md:21]: Definitely.
Those are some glitches that may be introduced when I renamed `tarball` node.
Need to fix this.

Comment (JenySadadia) [reply doc/pipeline-details.md:21]: I have created it with mermaid live editor and the block you mentioned is in `tarball` service itself.
I wonder why it is different on staging.
Attaching the flowchart from https://mermaid.live

image

File: doc/pipeline-details.md (hunk 1 above)
Comment (JenySadadia) [reply doc/pipeline-details.md:17]: Yes, will fix this.

File: doc/pipeline-details.md (hunk 2 above)
Comment (JenySadadia) [reply doc/pipeline-details.md:22]: Even https://github.com/kernelci/kernelci-api/blob/staging.kernelci.org/doc/pipeline-details.md matches with the flowchart I attached.

**Review by gctucker — APPROVED** [review]

**Pull request merged**